src/three-components/TextureManager.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TextureLoader, EventDispatcher} from 'three';
+import EquirectangularToCubemap from '../third_party/three.equirectangular-to-cubemap/EquirectangularToCubemap.js';
+import EnvMapGenerator from './EnvMapGenerator.js';
+
+const loader = new TextureLoader();
+
+export default class TextureManager extends EventDispatcher {
```

Comment:
Naming nit: I expect a manager to actually manage its namesake. This class doesn't seem to manage textures, but rather seems scoped to texture conversion. The name "manager" would seem more appropriate if the TextureManager also handled disposing of the textures it generates at the appropriate time, but it doesn't seem to do that.

Reply:
Agreed -- unfortunately due to how textures work, it'd be difficult to manage their lifetimes here while giving out textures -- there could be a solution in the future where we have our own texture class that allows individual subscribers to relinquish the texture, and only after all consumers have relinquished does it actually dispose (if we were sharing resources).

TextureGenerator?

Reply:
WFM